theano/gpuarray/tests/run_dnn_conv.py

```diff
@@ -0,0 +1,153 @@
+                                FLOAT_CONFIG=FLOAT_CONFIG, DOUBLE_CONFIG=DOUBLE_CONFIG)
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument('computation', choices=computations,
+                    help='Computation to run.')
+
+parser.add_argument('-a', '--algo', choices=algorithms, required=True,
+                    help='Algorithm to use for computation.')
+parser.add_argument('-i', '--input-shape', action=TupleAction, required=True,
+                    help='Input shape. Comma-separated list of integers (no spaces).')
+parser.add_argument('-f', '--filter-shape', action=TupleAction, required=True,
+                    help='Filter shape. Comma-separated list of integers (no spaces).')
+
+parser.add_argument('-D', '--dtype-config', choices=list(sorted(data_type_configurations.keys())), default=None,
+                    help='Data type configuration for (data type; precision). Default (theano floatX; theano floatX). '
+                         'To specify data type configuration, you can either use this option or set data type and '
+                         'precision separately with "-t" and "-p" options.')
+parser.add_argument('-t', '--dtype', choices=types, default=None,
+                    help='Data type (default theano floatX).')
+parser.add_argument('-p', '--precision', choices=types, default=None,
+                    help='Precision (default theano floatX).')
+parser.add_argument('-s', '--subsample', action=TupleAction,
+                    help='Subsample. Comma-separated list of integers (no spaces). '
+                         'Default: 1 per dimension.')
+parser.add_argument('-d', '--dilation', action=TupleAction,
+                    help='Dilation. Comma-separated list of integers (no spaces). '
+                         'Default: 1 per dimension.')
+parser.add_argument('-b', '--border-mode', default='valid', action=BorderAction,
+                    help='Border mode. "valid" (default), "full", "half" '
+                         'or a comma-separated list of integers (no spaces).')
+parser.add_argument('-c', '--conv-mode', choices=('conv', 'cross'), default='conv',
+                    help='Conv mode (default: conv).')
+parser.add_argument('-A', '--alpha', type=float, default=1,
+                    help="alpha (floating), must not be zero. Default 1.")
+parser.add_argument('-B', '--beta', type=float, default=0,
+                    help='beta (floating). Default 0.')
+
+parser.add_argument('-I', '--print-infos', action='store_true', default=False,
+                    help='Print some infos before testing.')
+
+args = parser.parse_args(args)
+
+test = args.computation
+if len(args.input_shape) != len(args.filter_shape):
+    raise ValueError('Expected same length for input shape and filter shape')
+if len(args.input_shape) not in (4, 5):
+    raise ValueError('Expected length 4 or 5 for input shape')
+ndim = len(args.input_shape) - 2
+if ndim == 2:
+    tests = TestDnnConv2D()
+if ndim == 3:
```

Comment:
use "elif" when it make sence instead of "if" in the file. It make it more readable my making it clear there are condition that won't happen together.

Reply:
Fixed